### PR TITLE
Bump Ubuntu from 20.04 to 22.04

### PR DESCRIPTION
**Pull request by leotm**
Ref: https://github.com/leotm/react-native-template-new-architecture/issues/918

TODO: Update

- [x] https://github.com/react-native-community/docker-android/blob/master/.github/workflows/build.yaml
- [x] https://github.com/react-native-community/docker-android/blob/master/Dockerfile
- [x] https://github.com/react-native-community/docker-android/blob/master/.github/workflows/publish.yml
- [x] Code review: https://github.com/react-native-community/docker-android/pull/179#discussion_r954621280

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # build buck from source
-FROM ubuntu:20.04 AS buck
```
Comment (gengjiawen) [inline Dockerfile:2]: Not necessary, see https://github.com/react-native-community/docker-android/pull/178.
Can you only keep the yml change ?

Comment (leotm) [reply Dockerfile]: Gotcha thx ^ sure thing, done: https://github.com/react-native-community/docker-android/pull/179/commits/7a750482b9ca6777996ba16f5673b23c0f8a6c77

**Review by gengjiawen — APPROVED** [review]

**Comment by gengjiawen** [comment]
Thx for the contribution.

**Comment by shamilovtim** [comment]
Hmm ubuntu isn't actually bumped in the Dockerfile, just on this project's CI example app?

https://github.com/react-native-community/docker-android/blob/master/Dockerfile#L1

**Comment by cortinico** [comment]
> Hmm ubuntu isn't actually bumped in the Dockerfile, just on this project's CI example app?

Feel free to send a PR for the bump. You'll need to handle some packages that have been removed in Ubuntu 22.04 though so that's non trivial.

**Pull request merged**